### DISCUSSION TITLE
Use flutter_embed_unity with Flutter 3.7.10

### DISCUSSION THREAD

**Discussion by moaznarsun** [learntoflutter/flutter_embed_unity#26] [Development]
Huy, I have to use this sdk with flutter 3.7.10 due to some limitations. I saw in the documentation that it is supported but when i follow the steps I get this Error. Is there any work around for this?
Please help with it as soon as possible.
![image](https://github.com/user-attachments/assets/8601e6b3-267c-470e-b8e3-e8a38bc3a3c7)

**Comment by jamesncl** [comment]
Yep you're right, it looks like the Dart constraint on the package was set unnecessarily low, which was incompatible with earlier versions of Flutter. I've just published a new version (1.2.2) with the minimum Dart set to 2.18 or higher, so updating the version of flutter_embed_unity to 1.2.2 should fix your problem:
```yaml
dependencies:
  flutter_embed_unity: ^1.2.2
```

**Reply by moaznarsun** [reply]
Hey thanks. I tried it but it is still giving me the same error.  I think you have lowered the sdk requirement for flutter_embed_unity_2022_3_android but not for flutter_embed_unity. Can you please verify?
This is how I have added the dependencies.
![image](https://github.com/user-attachments/assets/b8b83975-b7f7-4072-bcbc-01f2857fa0d8)

**Comment by jamesncl** [comment]
Ah my bad, you're right I forgot to change the constraint in the app-facing package. Just published 1.2.3! Let me know if that works :)

```
dependencies:
  flutter_embed_unity: ^1.2.3
```

PS you don't need to include flutter_embed_unity_2022_3_android in your pubspec.yaml unless you have a specific need to override the default version, as it will automatically be included as part of flutter_embed_unity with the version specified in [its pubspec.yaml](https://github.com/learntoflutter/flutter_embed_unity/blob/main/flutter_embed_unity/pubspec.yaml) (currently flutter_embed_unity_2022_3_android version 1.1.1). You only need to directly add flutter_embed_unity_2022_3_android to your pubspec.yaml if you need to override the default for some reason. This will come in useful in the future when Unity 6 is officially released, as @timbotimbo has already started work on a separate platform package `flutter_embed_unity_6000_android`
